### ISSUE TITLE
[java] Possible False Positive (Unnecessary wrapper)

### ISSUE THREAD

**Issue by amarkevich** [pmd/pmd#2061]
**Affects PMD Version:** 3.12.0

**Rule:** UnnecessaryWrapperObjectCreation

**Description:** warns about PMD.UnnecessaryWrapperObjectCreation

**Code Sample demonstrating the issue:**

```
        Object actual = "1";
        org.junit.Assert.assertEquals(1L, Long.valueOf(actual.toString()).longValue());
```

**Running PMD through:** *Maven*

`longValue()` required for `assertEquals(long expected, long actual)` method; otherwise `assertEquals(Object expected, Object actual)` used

**Comment by jsotuyod** [comment]
@amarkevich I don't think this is a false positive at all, the rule is flagging the `valueOf` (creating a boxed primitive) + `longValue` (unboxing) usage, in favor of simply using `parseLong` which makes no allocations. Your code could be rewritten as:

```java
Object actual = "1";
org.junit.Assert.assertEquals(1L, Long.parseLong(actual.toString()));
```
